src/Resources/Timeline.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+namespace SevenShores\Hubspot\Resources;
+
+use DateTime;
+
+class Timeline extends Resource
+{
+    /**
+     * Create or Update Timeline Event
+     *
+     * @param int           $appId
+     * @param int           $eventTypeId
+     * @param string        $id
+     * @param int|null      $objectId
+     * @param string|null   $email
+     * @param string|null   $utk
+     * @param array         $extraData
+     * @param DateTime|null $timestamp
+     *
+     * @return mixed
+     *
+     * @see http://developers.hubspot.com/docs/methods/timeline/create-or-update-event
+     */
+    public function createOrUpdate(
+        $appId,
+        $eventTypeId,
+        $id,
+        $objectId = null,
+        $email = null,
+        $utk = null,
+        $extraData = [],
+        DateTime $timestamp = null
+    ) {
+        $endpoint = "https://api.hubapi.com/integrations/v1/{$appId}/timeline/event";
+
+        $data['json'] = [
+            'eventTypeId' => $eventTypeId,
+            'id'          => $id,
+            'objectId'    => $objectId,
+            'email'       => $email,
+            'utk'         => $utk,
+            'extraData'   => $extraData,
+        ];
+
+        if (isset($timestamp)) {
+            // Times must be in milliseconds epoch time.
+            $data['json']['timestamp'] = $timestamp->getTimestamp() * 1000;
+        }
```

Comment:
Not sure if I like the DateTime typehinting. If I see the variable name I am going to expect to send an int, but then it's a DateTime. Also, what if i am working with data I got from hubspot and it is already milliseconds?

How about something like: 

```php
if (is_numeric($timestamp) && strlen((string)$num) == 10) {
    $timestamp = $timestamp * 1000;
}
```